### PR TITLE
SES6: qa: suse/tier2/drive-replacement: use dynamic value for drive group limit

### PR DESCRIPTION
**Pull request by shyukri**
Also make flake8 happy

Signed-off-by: Shyukri Shyukriev <shshyukriev@suse.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available @susebot commands</summary>

- `@susebot run make check`
- `@susebot run make check sles`
- `@susebot run make check leap`

</details>

### CODE REVIEW

**Comment by shyukri** [comment]
Test now passes on Teuthidas: http://10.162.230.4/teuthology/worker-2020-08-25_20:21:00-suse:tier2:drive-replacement-ses6-none-basic-nue-teuthida/1745/teuthology.log and openstack: http://teuth-ses.prv.suse.net:8081/runner-2020-08-25_19:41:14-suse:tier2:drive-replacement-ses6-none-basic-ecp/

File: qa/tasks/deepsea.py [hunk 1]
```diff
@@ -1908,6 +1908,28 @@ def wait_for_health_ok(self, **kwargs):
                                "after reaching timeout"
                                .format(cluster_status))
 
+    def drive_group_modify_limit_by(self, **kwargs):
```
Comment (kshtsk) [inline qa/tasks/deepsea.py:1911]: Can we change it `set_drive_group_limit`.

File: qa/tasks/util/__init__.py [hunk 2]
```diff
@@ -179,8 +180,14 @@ def introspect_roles(ctx, logger, quiet=True):
...
-    nodes_random_storage = random.sample(nodes_storage, 2 if len(nodes_storage) > 1 else 1)
-    nodes_random_monitor = random.sample(nodes_monitor, 2 if len(nodes_monitor) > 1 else 1)
+    nodes_random_storage = random.sample(  # noqa: F841
+                            nodes_storage,
+                            2 if len(nodes_storage) > 1 else 1
+                            )
+    nodes_random_monitor = random.sample(  # noqa: F841
+                            nodes_monitor,
+                            2 if len(nodes_monitor) > 1 else 1
+                            )
```
Comment (kshtsk) [inline qa/tasks/util/__init__.py:190]: Why this is required?

File: qa/tasks/util/__init__.py [hunk 3]
```diff
@@ -10,6 +10,7 @@
     )
 from paramiko.ssh_exception import NoValidConnectionsError
 
+
```
Comment (kshtsk) [inline qa/tasks/util/__init__.py:13]: This is not related to the patch.

File: qa/tasks/util/__init__.py [hunk 4]
```diff
@@ -216,7 +223,7 @@ def introspect_roles(ctx, logger, quiet=True):
     elif len(nodes_monitor) < 3:
         ctx['dev_env'] = True
     else:
-        ctx['dev_env'] = False    
+        ctx['dev_env'] = False
```
Comment (kshtsk) [inline qa/tasks/util/__init__.py:226]: Could you gather all these changes not related to the issue you're fixing to a separate commit.

File: qa/suites/deepsea/tier2/drive-replacement/1-deepsea.yaml [hunk 5]
```diff
@@ -6,12 +6,15 @@ tasks:
                         target: '*'
                         data_devices:
                             all: true
-                            limit: 4
+                            limit: 99 # this is altered with dynamic value before stage.2
```
Comment (kshtsk) [inline qa/suites/deepsea/tier2/drive-replacement/1-deepsea.yaml:9]: How about to use: negative valumes for the limit here to use as number of reserved disk?

File: qa/tasks/deepsea.py [hunk 6]
```diff
@@ -1908,6 +1908,28 @@ def wait_for_health_ok(self, **kwargs):
...
+        drive_group_modify_limit_by:
+            limit: -1
+
+        this will take all number of disks, reduce it by OS reserved
+        and then reduce/add by the limit value
+
+        Used to cover scenarios for drive-replacement test on OpenStack and HW
+        """
+        number_of_disks = self.master_remote.sh("sudo lsblk -d -n | wc -l")
+        os_reserved_disk = 1
```
Comment (kshtsk) [inline qa/tasks/deepsea.py:1925]: this is a boot disk

File: qa/tasks/deepsea.py [hunk 7]
```diff
@@ -1908,6 +1908,28 @@ def wait_for_health_ok(self, **kwargs):
...
+        available_disks = int(number_of_disks) - os_reserved_disk
+        value = list(kwargs.values())[0]
+        limit = available_disks + value
+        file = "/srv/salt/ceph/configuration/files/drive_groups.yml"
```
Comment (kshtsk) [inline qa/tasks/deepsea.py:1929]: x=remote.read_file()
y=yaml.safe_load(x)
y.set('limit', n)
remote.write_file(y.safe_dump(x))

File: qa/tasks/deepsea.py [hunk 8]
```diff
@@ -1908,6 +1908,28 @@ def wait_for_health_ok(self, **kwargs):
...
+        Helper to modify drive_groups limit field value
+
+        ex.
+        drive_group_modify_limit_by:
+            limit: -1
+
+        this will take all number of disks, reduce it by OS reserved
+        and then reduce/add by the limit value
+
+        Used to cover scenarios for drive-replacement test on OpenStack and HW
```
Comment (kshtsk) [inline qa/tasks/deepsea.py:1922]: Drop this line.

File: qa/tasks/deepsea.py (hunk 6 above)
Comment (shyukri) [reply qa/tasks/deepsea.py]: Dropped. WIll be handled by `ceph-volume inventory` output

File: qa/tasks/util/__init__.py (hunk 4 above)
Comment (shyukri) [reply qa/tasks/util/__init__.py:220]: Done

**Comment by smithfarm** [comment]
@shyukri Could you squash the commits and prefix the commit message with `qa/deepsea: ` please

**Comment by ryg-** [comment]
Could you please do tests on teuthidas?

File: qa/tasks/util/__init__.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (shyukri) [reply qa/tasks/util/__init__.py:13]: yes, according to flake8: `qa/tasks/util/__init__.py:13:1: E302 expected 2 blank lines, found 1`

**Review by smithfarm — APPROVED** [review]
LGTM. Please merge once you're confident it won't break any CI testing environments.

**Comment by ryg-** [comment]
http://teuthology.ses.suse.de:8081/worker-2020-09-09_08:08:40-suse:tier2-ses6-none-basic-nue-teuthida/
LGTM

**Comment by smithfarm** [comment]
Thanks @shyukri. I also cherry-picked it to `ses6-downstream-commits` and `ses6-deepsea-pr`.
